File: EDSR.py
import cv2
import matplotlib.pyplot as plt
# Read image
img = cv2.imread("AI-Courses-By-OpenCV-Github.jpg")
plt.imshow(img[:,:,::-1])

# Cropout OpenCV logo
img = img[:80,850:]
#plt.imshow(img[:,:,::-1])
#plt.show()

sr = cv2.dnn_superres.DnnSuperResImpl_create()

path = "EDSR_x4.pb"

sr.readModel(path)

sr.setModel("edsr",4)

result = sr.upsample(img)

# Resized image
resized = cv2.resize(img,dsize=None,fx=4,fy=4)

#plt.figure(figsize=(12,8))
#plt.subplot(1,3,1)
# Original image
#plt.imshow(img[:,:,::-1])
#plt.subplot(1,3,2)
# SR upscaled
#plt.imshow(result[:,:,::-1])
#plt.subplot(1,3,3)
# OpenCV upscaled
#plt.imshow(resized[:,:,::-1])
#plt.show()

File: EDSR.cpp
#include <iostream>
#include <opencv2/dnn_superres.hpp>
#include <opencv2/imgproc.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;
using namespace dnn;
using namespace dnn_superres;

Mat upscaleImage(Mat img, string modelName, string modelPath, int scale){
	DnnSuperResImpl sr;
	sr.readModel(modelPath);
	sr.setModel(modelName,scale);
	// Output image
	Mat outputImage;
	sr.upsample(img, outputImage);
	return outputImage;
}

int main(int argc, char *argv[])
{
	// Read image
	Mat img = imread("AI-Courses-By-OpenCV-Github.jpg");
	
	// Region to crop
	Rect roi;
	roi.x = 850;
	roi.y = 0;
	roi.width = img.size().width - 850;
	roi.height = 80;
	img = img(roi);

	// EDSR (x4)
	string path = "EDSR_x4.pb";
	string modelName = "edsr";
	int scale = 4;
	Mat result = upscaleImage(img, modelName, path, scale);

	// Image resized using OpenCV
	Mat resized;
	cv::resize(img, resized, cv::Size(), scale, scale);
   

	/*imshow("Original image",img);
	imshow("SR upscaled",result);
	imshow("OpenCV upscaled",resized);
	waitKey(0);
	destroyAllWindows();*/

	return 0;
}